database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -212,11 +244,36 @@ public void write(DataOutput out) throws IOException {
    * @param stmt the {@link PreparedStatement} to write the {@link StructuredRecord} to
    */
   public void write(PreparedStatement stmt) throws SQLException {
+    List<String> updatedKeyList = new ArrayList<>();
+    if (relationTableKey != null) {
+      updatedKeyList = Arrays.asList(relationTableKey.split(","));
+    }
+    modifiableColumnTypes = new ArrayList<>(columnTypes);
     for (int i = 0; i < columnTypes.size(); i++) {
       ColumnType columnType = columnTypes.get(i);
       Schema.Field field = record.getSchema().getField(columnType.getName());
       writeToDB(stmt, field, i);
+      if (fillUpdateParams(updatedKeyList, columnType)) {
+        modifiableColumnTypes.add(columnType);
+      }
     }
+    if (operationName != null && relationTableKey != null) {
+      // index to store the latest place of question mark to be allocated
+      int j = columnTypes.size();
+      for (int i = j; i < modifiableColumnTypes.size(); i++) {
+        ColumnType columnType = modifiableColumnTypes.get(i);
+        Schema.Field field = record.getSchema().getField(columnType.getName());
+        writeToDB(stmt, field, i);
+      }
+    }
+  }
+
+  private boolean fillUpdateParams(List<String> updatedKeyList, ColumnType columnType) {
+    if (operationName != null && operationName.equalsIgnoreCase("UPDATE") &&
```

Comment:
instead of 'update' use already defined constant from ETLDBOutputFormat class.

Reply:
done